Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -259,74 +259,93 @@ namespace AzToolsFramework
                 m_instanceToTemplateInterface->GenerateEntityDomBySerializing(
                     newContainerEntityDomInitialStateWithTransform, *newContainerEntity);
 
-                // Parent the non-container top level entities to the new container entity.
-                // It also patches the entity DOMs to the new template without undo/redo support.
-                EditorEntityContextNotificationBus::Broadcast(&EditorEntityContextNotification::SetForceAddEntitiesToBackFlag, true);
-                for (AZ::Entity* topLevelEntity : topLevelEntities)
+                // Helper function to create a link between a nested instance and the newly created instance
+                auto createLinkForNestedInstanceFunc = [&](Instance& nestedInstance)
```

Comment:
nit: I recommend renaming it to 'createLinkForNestedInstance' to avoid using half names.